netx/httptransport/http3transport_test.go

```diff
@@ -2,18 +2,25 @@ package httptransport_test
 
 import (
 	"context"
+	"crypto/tls"
 	"errors"
 	"net/http"
 	"strings"
 	"testing"
 
+	"github.com/lucas-clemente/quic-go"
 	"github.com/ooni/probe-engine/netx/dialer"
 	"github.com/ooni/probe-engine/netx/httptransport"
 	"github.com/ooni/probe-engine/netx/selfcensor"
 )
 
+type MockHTTP3Dialer struct{}
+
+func (d MockHTTP3Dialer) Dial(network, host string, tlsCfg *tls.Config, cfg *quic.Config) (quic.EarlySession, error) {
+	return quic.DialAddrEarly(host, tlsCfg, cfg)
+}
 func TestUnitHTTP3TransportSuccess(t *testing.T) {
-	txp := httptransport.NewHTTP3Transport(selfcensor.SystemDialer{}, dialer.TLSDialer{})
+	txp := httptransport.NewHTTP3Transport(httptransport.Config{selfcensor.SystemDialer{}, MockHTTP3Dialer{}, dialer.TLSDialer{}})
```

Comment:
Could you explicitly initialise fields? I think this makes the code more obvious.

(I.e., `X{A: a, B: b}` rather then `X{a, b}`)

The motivation of this request is that adding explicit initialisation of fields allows us to take the liberty of adding new fields and changing the order of fields in the `Config` structure w/o breaking the build.